custom_components/rct_power/lib/api.py

```diff
@@ -149,15 +146,9 @@ async def _read_object(
                             )
                             continue
 
-                        decoded_value: (
-                            bool
-                            | bytes
-                            | float
-                            | int
-                            | str
-                            | tuple[datetime, dict[datetime, int]]
-                            | tuple[datetime, dict[datetime, EventEntry]]
-                        ) = decode_value(data_type, response_frame.data)  # type: ignore
+                        decoded_value: ApiResponseValue = decode_value(
```

Comment:
After this change the type checker complains that the type is partially unknown, which was the reason why I added these type annotations. Any idea how to resolve that elegantly?

Reply:
The `partially unknown` issue was there before as well. It's the `Argument of type "DataType" cannot be assigned` error that was "uncovered". My intention here was to just replace the type definition with the existing type alias, without changing the type ignore. Seems like the ruff format broke that a bit.

Added an additional `type ignore` comment for the issue mentioned earlier.

Reply:
> At some point this might need to be fixed upstream in `rctclient` library.

agreed... maybe I should create a fork for use until the original author has time to revive the repo